src/debug.h

```diff
@@ -63,44 +63,38 @@ std::string GetDebugString();
 /* Shorter form for passing filename and linenumber */
 #define FILE_LINE __FILE__, __LINE__
 
-/**
- * Used for profiling.
- *
+/** TicToc profiling.
  * Usage:
- * TIC();
- *   --Do your code--
- * TOC("A name", 1);
- *
- * When you run the TIC() / TOC() multiple times, you can increase the '1'
- *  to only display average stats every N values. Some things to know:
- *
- * for (int i = 0; i < 5; i++) {
- *   TIC();
- *     --Do your code--
- *   TOC("A name", 5);
- * }
- *
- * Is the correct usage for multiple TIC() / TOC() calls.
- *
- * TIC() / TOC() creates its own block, so make sure not the mangle
- *  it with another block.
- *
- * The output is counted in microseconds. Mainly useful for local optimisations.
- **/
-#define TIC() {\
-	auto _start_ = std::chrono::high_resolution_clock::now();\
-	static uint64_t _sum_ = 0;\
-	static uint32_t _i_ = 0;
-
-#define TOC(str, _count_)\
-	_sum_ += (std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::high_resolution_clock::now() - _start_)).count();\
-	if (++_i_ == _count_) {\
-		Debug(misc, 0, "[{}] {} us [avg: {:.1f} us]", str, _sum_, _sum_/(double)_i_);\
-		_i_ = 0;\
-		_sum_ = 0;\
-	}\
-}
+ * static TicToc::State state("A name", 1);
+ * TicToc tt(state);
```

Comment:
Hmm, did not expect a RAII with two parts. Why the state? Why not just `TicToc tt("a name", 1)`? (honest question, ofc)

Reply:
If there's no state you can't persist the state between runs when count is not 1.

Reply:
Lol, doh. Fair. You could have mentioned that somewhere, but okay :P :P Compensate for my lack of understanding! :P